client/src/App.js

```diff
@@ -84,15 +79,16 @@ function App() {
     }, 7000);
   }
 
-  function set_LoggedUser(value) {
-    setLoggedUser(value);
-  }
+  useEffect(() => {
```

Comment:
I tried to understand here. Because you set the loggedIn to true inside the local storage as soon as successfully logged in or registered and then direct to the user to the chatroom. TriggerRerender doesn't affect people can login or register based on what my testing. I deleted these few lines of codes and after that I still can render or login or register successfully without them.

Reply:
@Junjpan I was having issues having the page render again.  I had it working with login, but with register, it wouldn't switch and it seemed like this was the only solution that seemed to work.  Are you saying you deleted the `useEffect` and it worked?

Reply:
@rusye Yes, After I deleted the codes below, I still can go to the chatroom from either login or register component.
useEffect(()=>{
if (localStorage.loggedIn) {
      set_new_user();
    }
  }, [triggerRerender]);
}

Reply:
That's interesting, I'll give it a try

Reply:
@Junjpan It works, but now I get this error in the console `Line 65:10:  'triggerRerender' is assigned a value but never used no-unused-vars`, I don't know where else to use that state.

Reply:
Can you delete line 65, if it works when you delete the `useEffect` beginning on line 82?

Reply:
@rusye It looks like if you deleted the line 65, will cause register component not working properly. 
Anyway, why we don't just stick to our original way using function set_LoggedUser(value) {
    setLoggedUser(value);
  } to control login and logout. To be honest, I prefer not to put the loggedIn Boolean inside the localstorage to control our login.

Reply:
@Junjpan The state changes when you refresh the page and the user will have to login again, which can be annoying 

@a-w-m the component won't rerender, which means the user has to manually refresh the page

Reply:
@rusye Gotcha. I guess for now we can use your way moving forward and we can figure out a better way when we have the time unless @a-w-m or @Igor333m have a better solution.